modules/series-service-impl/src/main/java/org/opencastproject/series/impl/SeriesServiceImpl.java

```diff
@@ -504,53 +504,78 @@ public void repopulate() throws IndexRebuildException {
+            if (currentOrg.equals(lastOrg)) {
+              seriesBundle.add(databaseSeries.get(current + i));
+            } else {
+              lastOrg = currentOrg;
+              break;
+            }
+          }
+          current += seriesBundle.size();
+
+          Organization organization = orgDirectory.getOrganization(seriesBundle.get(0).getOrganization());
+          User systemUser = SecurityUtil.createSystemUser(systemUserName, organization);
+          SecurityUtil.runAs(securityService, organization, systemUser,
+                  () -> {
+                    List<String> seriesIds = new ArrayList<>();
+                    List<Function<Optional<Series>, Optional<Series>>[]> updateFunctionBundle = new ArrayList<>();
+                    for (SeriesEntity series: seriesBundle) {
+                      String seriesId = series.getSeriesId();
+                      seriesIds.add(seriesId);
+                      logger.trace("Adding series {} for organization {} to the {} index.", seriesId,
+                              series.getOrganization(), index.getIndexName());
+                      List<Function<Optional<Series>, Optional<Series>>> updateFunctions = new ArrayList<>();
+                      DublinCoreCatalog catalog;
+                      try {
+                        catalog = DublinCoreXmlFormat.read(series.getDublinCoreXML());
+                        updateFunctions.add(getMetadataUpdateFunction(seriesId, catalog, organization.getId()));
+                      } catch (IOException | ParserConfigurationException | SAXException e) {
+                        logger.error("Could not read dublincore XML of series {}.", seriesId, e);
+                        return;
+                      }
+
+                      String aclStr = series.getAccessControl();
+                      if (StringUtils.isNotBlank(aclStr)) {
+                        try {
+                          AccessControlList acl = AccessControlParser.parseAcl(aclStr);
+                          updateFunctions.add(getAclUpdateFunction(seriesId, acl, organization.getId()));
+                        } catch (Exception ex) {
+                          logger.error("Unable to parse ACL of series {}.", seriesId, ex);
+                        }
+                      }
+
+                      try {
+                        Map<String, String> properties = persistence.getSeriesProperties(seriesId);
+                        updateFunctions.add(getThemePropertyUpdateFunction(seriesId,
+                                Optional.ofNullable(properties.get(THEME_PROPERTY_NAME)), organization.getId()));
+                      } catch (NotFoundException | SeriesServiceDatabaseException e) {
+                        logger.error("Error reading properties of series {}", seriesId, e);
+                      }
+                      updateFunctionBundle.add(updateFunctions.toArray(new Function[0]));
```

Comment:
<del>You pass on these three update functions as three entries. Later on, that means, you get the series from the index multiple times to then run an update instead of having one update per series.</del>

Never mind. I misread that as `addAll`, you do exactly that in the next function. But why not do it in the first place? Why the array structure in between?